### PR TITLE
Fix GetLocalApiUrl for cases with https enabled

### PR DESCRIPTION
**Pull request by JustAMan**
This should fix LiveTV and tuners

**Changes**

**Issues**

Probably fixes https://github.com/jellyfin/jellyfin/issues/2568 (cannot verify as I don't have a tuner)

### CODE REVIEW

**Comment by JustAMan** [comment]
Hey @Bond-009 it was you who broke that (as usual) :D here - https://github.com/jellyfin/jellyfin/pull/2221

File: Emby.Server.Implementations/ApplicationHost.cs [hunk 1]
```diff
@@ -1507,18 +1507,10 @@ public string GetLocalApiUrl(IPAddress ipAddress)
...
-            {
-                url.Append("http://");
-            }
-
-            url.Append(host)
+            url.Append(EnableHttps ? "https://" : "http://")
+                .Append(host)
                 .Append(':')
-                .Append(HttpPort);
+                .Append(EnableHttps ? HttpsPort : HttpPort);
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/ApplicationHost.cs:1513]: You should also check `Certificate != null`, you can see the condition checked to enable HTTPS here: https://github.com/jellyfin/jellyfin/blob/master/Jellyfin.Server/Program.cs#L277

Comment (JustAMan) [reply Emby.Server.Implementations/ApplicationHost.cs:1513]: while I agree, this `appHost.EnableHttps` is used throughout the codebase without checking for certificate... what's the best way of tackling this?

Comment (Bond-009) [reply Emby.Server.Implementations/ApplicationHost.cs:1513]: Create a new property `IsHttpsEnabled` that returns `EnableHttps && Certificate != null`?
And possibly remove `EnableHttps` from apphost and only have it in the config

Comment (JustAMan) [reply Emby.Server.Implementations/ApplicationHost.cs:1513]: I wanted to make that property, but I fear a lot of stuff would break... but let's try :)

Comment (JustAMan) [reply Emby.Server.Implementations/ApplicationHost.cs:1513]: Hmmm....
```cs
        public bool EnableHttps => SupportsHttps && ServerConfigurationManager.Configuration.EnableHttps;

        public bool SupportsHttps => Certificate != null || ServerConfigurationManager.Configuration.IsBehindProxy;
```
it's already there (not there only if "handled by reverse proxy" is enabled, but assuming https in local api url is in line with what other things are doing anyway).

Comment (mark-monteiro) [reply Emby.Server.Implementations/ApplicationHost.cs:1513]: Ah, good catch. Those properties are confusing -_-

**Pull request merged**